src/yaml.rs

```diff
@@ -204,19 +219,70 @@ impl YamlLoader {
     /// The `source` is interpreted as YAML documents and is parsed. Parsing succeeds if and only
     /// if all documents are parsed successfully. An error in a latter document prevents the former
     /// from being returned.
+    /// # Errors
+    /// Returns `ScanError` when loading fails.
     pub fn load_from_iter<I: Iterator<Item = char>>(source: I) -> Result<Vec<Yaml>, ScanError> {
-        let mut loader = YamlLoader {
-            docs: Vec::new(),
-            doc_stack: Vec::new(),
-            key_stack: Vec::new(),
-            anchor_map: BTreeMap::new(),
-        };
+        let mut loader = YamlLoader::default();
         let mut parser = Parser::new(source);
         parser.load(&mut loader, true)?;
         Ok(loader.docs)
     }
 }
 
+pub struct YamlDecoder<T: std::io::Read> {
```

Comment:
Could there be a doccomment for `YamlDecoder`?

Reply:
Thanks for the thorough review. I added docs and I think I've addressed all of the review notes.